tests/neg/given-pattern.scala

```diff
@@ -4,11 +4,11 @@ class Test {
   import scala.collection.immutable.{TreeSet, HashSet}
 
   def f2[T](x: Ordering[T]) = {
-    val (given y: Ordering[T]) = x
+    val (given Ordering[T]) = x
```

Comment:
Interesting. This allows for a given to be bound to a `val` which we could not do with `given y as Ordering[T] = x`
```scala
val (y as given Ordering[T]) = x
```

Reply:
If I understand this correctly, the y is _not_ implicit here (hence the error on next line), and the given Ordering is not visible outside of the pattern its part of, so while this syntax is permitted because any pattern can be used in a val, it's useless.

Reply:
@smarter: correct. implicits in patterns behave the same way.